src/tree/updater_gpu_hist.cu

```diff
@@ -224,6 +227,53 @@ struct CalcWeightTrainParam {
         learning_rate(p.learning_rate) {}
 };
 
+// index of the first element in cuts greater than v, or n none;
```

Comment:
I use something like this in the GPU predictor, we should look at standardising this function in device helpers. Not necessary for this PR though.